### ISSUE TITLE
sql/parser:   EXPLAIN does not pretty-print its AST correctly

### ISSUE THREAD

**Issue by knz** [cockroachdb/cockroach#51830]
Found by the RSG at https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTests/2107812?showLog=2107812_12614_5028.12614

```
        TestRandomSyntaxGeneration: rsg_test.go:756: Parse followed by Format is not idempotent: "EXPLAIN ANALYZE ( PLAN ) RESTORE FROM ( PLACEHOLDER ) WITH OPTIONS ( 'string' , EXPERIMENTAL_AUDIT )" -> "EXPLAIN ANALYZE RESTORE FROM 'placeholder' WITH \"string\", experimental_audit" != "EXPLAIN ANALYZE (DISTSQL) RESTORE FROM 'placeholder' WITH \"string\", experimental_audit"
```

cc @dt @miretskiy for triage

**Comment by dt** [comment]
Isn’t the printed RESTORE AST the same in both of those?

**Comment by knz** [comment]
Oh you're right. This is not a bulk I/O problem. The problem is in the EXPLAIN options.

@RaduBerinde @yuzefovich can you re-route this to the right person thanks.

**Comment by rohany** [comment]
I thought that @RaduBerinde fixed this already (or something that looked like this)

**Comment by RaduBerinde** [comment]
Indeed, should be fixed by #51477.
